guides/source/asset_pipeline.md

```diff
@@ -100,194 +76,289 @@ content into the name, usually at the end. For example a CSS file `global.css`
+
+# app/views/layouts/application.html.erb
+<%= javascript_importmap_tags %>
+
+# will include the following link before the importmap is setup:
+<link rel="modulepreload" href="https://ga.jspm.io/npm:@github/hotkey@1.4.4/dist/index.js">
+...
 ```
 
-Paths are traversed in the order they occur in the search path. By default,
-this means the files in `app/assets` take precedence, and will mask
-corresponding paths in `lib` and `vendor`.
+NOTE: Refer to [`importmap-rails`](https://github.com/rails/importmap-rails) repository for the most up-to-date documentation.
 
-It is important to note that files you want to reference outside a manifest must
-be added to the precompile array or they will not be available in the production
-environment.
+How to Use Sprockets
+-----------------------------
+
+In older versions of Rails, all assets were located in subdirectories of
+`public` folder such as `images` and `stylesheets`. Sprockets is designed to preprocess your assets stored in the configured directories and after processing expose them in the `public/assets` folder with configured fingerprinting and/or compression.
+
+Assets can still be placed in the `public` hierarchy. Any assets under `public`
+will be served as static files by the application or web server when
+[`config.public_file_server.enabled`][] is set to true. You must define `manifest.js` directives
+for files that must undergo some pre-processing before they are served.
+
+In production, Rails precompiles these files to `public/assets` by default. The
+precompiled copies are then served as static assets by the web server. The files
+in the `app/assets` are never served directly in production.
+
+[`config.public_file_server.enabled`]: configuring.html#config-public-file-server-enabled
+
+### Manifest Files and Directives
+
+When compiling assets with Sprockets, Sprockets needs to decide which top-level targets to compile, usually `application.css` and images. The top-level targets are defined in the Sprockets `manifest.js` file, by default it looks like this:
+
+```js
+//= link_tree ../images
+//= link_directory ../stylesheets .css
+//= link_tree ../../javascript .js
+//= link_tree ../../../vendor/javascript .js
+```
 
-[`config.assets.paths`]: configuring.html#config-assets-paths
+It contains _directives_ - instructions that tell Sprockets
+which files to require in order to build a single CSS or JavaScript file.
+
+This is meant to include the contents of all files found in the `./app/assets/images` directory or any subdirectories as well as any file recognized as JS directly at `./app/javascript` or `./vendor/javascript`.
+
+It will load any CSS from the `./app/assets/stylesheets` directory(not including subdirectories). Assuming that you have `application.css` and `marketing.css` files in the `./app/assets/stylesheets` folder it will allow you to load those stylesheets with `<%= stylesheet_link_tag "application" %>` or `<%= stylesheet_link_tag "marketing" %>` from your views.
```

Comment:
```suggestion
It will load any CSS from the `./app/assets/stylesheets` directory (not including subdirectories). Assuming that you have `application.css` and `marketing.css` files in the `./app/assets/stylesheets` folder it will allow you to load those stylesheets with `<%= stylesheet_link_tag "application" %>` or `<%= stylesheet_link_tag "marketing" %>` from your views.
```